hack/go-container.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/sh
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Simple posix sh reproducible go build container script with caching.
+#
+# Usage:
+#  hack/go.sh go version
+#  hack/go.sh go build -o /out/kind .
+set -o nounset -o errexit
+
+# ============================ SCRIPT SETTINGS =================================
+# get the repo root for defaulting OUT_DIR and SOURCE_DIR
+REPO_ROOT="${REPO_ROOT:-$(cd "$(dirname "$0")/.." && pwd)}"
+# output directory, will be mounted to /out, defaults to /bin in REPO_ROOT
+OUT_DIR="${OUT_DIR:-${REPO_ROOT}/build/_output/bin}"
```

Comment:
Is this what was in the original file? If not and you are adjusting it for this repo, you better do it in hack/go by exporting variables.